src/pages/api/signup.ts

```diff
@@ -0,0 +1,125 @@
+  message: string;
+};
+
+export type UserInput = {
+  firstName: string;
+  lastName: string;
+  email: string;
+};
+
+let isConnected = false;
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<SignUpResponse>
+) {
+  if (req.method === 'POST') {
+    try {
+      // Ensure database connection
+      if (!isConnected) {
+        await connectToDatabase();
+        isConnected = true;
+      }
+
+      const { firstName, lastName, email } = req.body as UserInput;
+
+      // Basic validation
+      if (!firstName || !lastName || !email) {
+        return res.status(400).json({ message: 'All fields are required' });
+      }
+      console.log(email);
+      if (!isValidEmail(email)) {
+        return res.status(400).json({ message: 'Invalid email format' });
+      }
+      console.log("is valid email", isValidEmail(email));
+      // Check if the user already exists
+      const existingUser = await User.findOne({ email });
+      console.log(existingUser,"existingUser");
+
+      if (existingUser) {
+        if (existingUser.isVerified) {
+          return res.status(400).json({ message: 'Email already registered and verified' });
+        } else {
+          // User exists but not verified, generate new OTP
+          const newOtp = generateOTP();
+          const newOtpExpires = new Date(Date.now() + 10 * 60 * 1000); // 10 minutes from now
+
+          // Update user with new OTP and expiration
+          existingUser.otp = newOtp;
+          existingUser.otpExpires = newOtpExpires;
+          await existingUser.save();
+
+          // Send new OTP via email
+          await sendOTPEmail(email, newOtp);//Todo later
+
+          return res.status(200).json({ message: 'New OTP sent. Please check your email for verification.' });
+        }
+      }
+
+      // Generate OTP
+      const otp = generateOTP();
+      console.log(otp,"otp");
+      const otpExpires = new Date(Date.now() + 10 * 60 * 1000); // 10 minutes from now
+
+      // Create new user
+      const newUser: IUser = new User({
+        email,
+        firstName,
+        lastName,
+        otp,
+        otpExpires,
+      });
+      console.log(newUser,"newUser");
+      // Save user to database
+      const newSavedUser=await newUser.save();
+      console.log(newSavedUser,"newSavedUser");
+      // Send OTP via email
+      console.log("otp sending...");
+      const isEmailSent=await sendOTPEmail(email, otp);
+      console.log(isEmailSent,"isEmailSent");
+      //Todo: send email
+      res.status(200).json({ message: 'User registered successfully. Please check your email for the OTP.' });
+    } catch (error) {
+      console.error('Error registering user:', error);
+      res.status(500).json({ message: 'Error registering user' });
+    }
+  } else {
+    res.setHeader('Allow', ['POST']);
+    res.status(405).json({ message: `Method ${req.method} Not Allowed` });
+  }
+}
+
+export function isValidEmail(email: string): boolean {
```

Comment:
this should be used in both login/signup. since this a common function. move this to helper.ts